### DISCUSSION TITLE
Refined Error Reporting

### DISCUSSION THREAD

**Discussion by jkHCBB** [CMSgov/price-transparency-guide-validator#110] [Ideas]
Hello team, I have a new feature request in mind

Current State:
When running MRF's through the validator,  only the first encountered error is reported and the file scan stops.

New Feature:
1.When there is more than one error in the file can we finish running the file through the validator and report the rest of the issues in the file as well? 

2. If the same error is encountered more than once, can we add a count to the message?
ex:
Error report:
{
    "required": {
        "missing": [
            "expiration_date"
        ],
        "errorCode": 15,
        "instanceRef": "#/in_network/0/negotiated_rates/0/negotiated_prices/0",
        "schemaRef": "#/definitions/negotiated_price"
    }
}
Error Name: required
****_insert an error count in the message_****
Message: (16) Objects are missing the following members required by the schema: 'expiration_date'.
**
Instance: #/in_network/0/negotiated_rates/0/negotiated_prices/0
Schema: #/definitions/negotiated_price

This would greatly reduce the amount of time spent debugging files with errors by drastically cutting down the amount of times the validator would need to be run during the debugging process. 

Thanks again,
Jake

**Comment by ethanLhciactive** [comment]
any update on this idea?

**Reply by jkHCBB** [reply]
Since I posted this updates were made and more info is presented. As far as if they will continue to build on that, I cant say, But am hopeful. 

In the mean time I built my own simple validator that tells me what I need.

**Reply by ethanLhciactive** [reply]
did you manage to have the file run through the validator for more than one error? I can't seem to get that solution

**Reply by jkHCBB** [reply]
I did but I used different code than what is in the CMS validator. I think the json schema package they use is the limiting factor on multi error reporting, but I don't remember exactly what I found at this point. If you really need to have the ability to see all the errors I could share what I made?

**Reply by ethanLhciactive** [reply]
if you could share it I would appreciate it greatly.

**Reply by jkHCBB** [reply]
https://github.com/jkHCBB/validator

I think I have everything in here. PM me if you need anything

**Reply by jkHCBB** [reply]
Cant promise its good code but it works. If you have any suggestions or want to contribute to it feel free to share!

**Reply by ethanLhciactive** [reply]
Thank you so much

**Reply by jkHCBB** [reply]
anytime
